pkg/operatorsource/prephase.go

```diff
@@ -0,0 +1,85 @@
+package operatorsource
+
+import (
+	"context"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/datastore"
+	"github.com/operator-framework/operator-marketplace/pkg/phase"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewPrePhaseReconciler returns a pre-phase Reconciler.
+func NewPrePhaseReconciler(logger *log.Entry, datastore datastore.Writer, client client.Client) Reconciler {
+	return &prePhaseReconciler{
+		logger:    logger,
+		datastore: datastore,
+		client:    client,
+	}
+}
+
+// prePhaseReconciler implements Reconciler interface.
```

Comment:
Does `OutOfSyncReconciler` make it better, since it detects if an `OperatorSource` is not in sync any longer? Feel free to suggest another name based on its primary role. :)